### PR TITLE
Updated dependencies and minor documentation change

### PR DESCRIPTION
**Pull request by dakkad**
Updated gradle dependencies and minor tweaks to the documentation to capture the current set of features supported by the app.

### CODE REVIEW

**Comment by dakkad** [comment]
Added compile options source and target comptibility.

File: mlkit/app/build.gradle [hunk 1]
```diff
@@ -26,6 +26,10 @@ android {
         exclude 'META-INF/androidx.exifinterface_exifinterface.version'
         exclude 'META-INF/proguard/androidx-annotations.pro'
     }
+    compileOptions {
```
Comment (samtstern) [inline mlkit/app/build.gradle]: Is this required for the mlkit sample app today?  Or is this required by the camera libraries?

**Comment by samtstern** [comment]
@dakkad thanks for the contribution!  Just one quesiton.

**Comment by samtstern** [comment]
@dakkad do you plan to continue with this PR?  If you're too busy I can try and copy your docs changes myself.

File: mlkit/app/build.gradle (hunk 1 above)
Comment (dakkad) [reply mlkit/app/build.gradle]: It is required for the current sample AFAIK.

**Comment by dakkad** [comment]
@samtstern I plan on leaving this PR as is and dealing with other changes separately rather than expanding the scope here.

**Comment by samtstern** [comment]
@dakkad I noticed you just pushed another commit to resolve the conflicts on this PR?  Do you still want to merge this?  If so my only remaining objection is the Java 8 upgrade, and I think we could easily remove that.

**Comment by samtstern** [comment]
Ok it's been a while without a response so I am going to close this PR.  Feel free to open a new one!

**Pull request closed without merging**